### PR TITLE
Recognize new maven, gradle, eclipse project(s) after first init

### PR DESCRIPTION
**Pull request by snjeza**
See https://github.com/eclipse/eclipse.jdt.ls/issues/144
Requires https://github.com/eclipse/eclipse.jdt.ls/pull/901

Signed-off-by: Snjezana Peco <snjezana.peco@redhat.com>

### CODE REVIEW

**Comment by jdneo** [comment]
Just an idea come up in my mind: how about prompt users to import if we detect the workspace contains valid project?

Take an example of the C# extension, when user opens a C# project folder, it will pop up a dialog to ask user if he want's to add required assets for build/debug:
Screen Shot 2020-03-13 at 9 49 35 AM

We can do the same thing, the Java Language Server can run in syntax mode by default, if it detects there is any build configuration files, then asks user to import it as a Java project. We can benefit from this experience by:

- Give users en expectation that importing might take some time to finish
- Won't generate `.project`, `.classpath` file by default, if the user just wants to view the code (#618)

**Comment by snjeza** [comment]
> Just an idea come up in my mind: how about prompt users to import if we detect the workspace contains valid project?

See https://github.com/redhat-developer/vscode-java/pull/746/commits/f2aa7af29e74a42264fa98710935f310cdaa3640#diff-04c6e90faac2675aa89e2176d2eec7d8R132

**Comment by fbricon** [comment]
It doesn't work: If  I have a multimodule Maven project, add a new module under the parent project, import doesn't find anything

**Comment by fbricon** [comment]
The succession of merge commits is just a huge mess.
Please rebase against master to have a simple commit.

**Comment by jdneo** [comment]
@fbricon The commits are rebased, could you please try again?

I tried both multi-module Maven/Gradle, both can detect new sub-module and generate `.classpath` and `.project`. The [test case](https://github.com/eclipse/eclipse.jdt.ls/blob/9f4bffbf81756d1c155d8633ed5b282890b92e0e/org.eclipse.jdt.ls.tests/src/org/eclipse/jdt/ls/core/internal/handlers/ImportNewProjectsTest.java) also imitates such scenario.

**Comment by jdneo** [comment]
@fbricon If still not working, is it possible to share the project to let me diagnostic?

**Comment by fbricon** [comment]
I opened an empty folder, then in vscode's terminal, I did `gradle init`, then tried to import the generated project:

Screen Shot 2020-05-18 at 6 55 54 PM

File: package.json [hunk 1]
```diff
@@ -553,6 +553,11 @@
         "title": "Update project configuration",
         "category": "Java"
       },
+      {
+        "command": "java.project.import",
+        "title": "Import Java projects in workspacce",
```
Comment (fbricon) [inline package.json:558]: workspace

**Comment by fbricon** [comment]
So this time, when calling the command, the extension activated, importing the project as part of the server startup, but ended up failing:  
Screen Shot 2020-05-19 at 3 00 12 PM

**Comment by fbricon** [comment]
I keep calling the import new project command, now the server is started, but still see the "No delegateCommandHandler for java.project.import" error

**Comment by fbricon** [comment]
Sorry,  looks like a PEBCAK. I didn't rebuild the proper server branch

**Pull request merged**